--- main_terminal.c
+++ main_terminal.c
@@ -153,7 +153,7 @@
 {
 }
 
-int running = 1;
+volatile int running = 1;
 
 void handleSignal(int signal)
 {
@@ -171,7 +171,10 @@
 
   timeStart = getTime();
 
-  input_init(SMALLINPUT_MODE_NORMAL,0,0);
+  if (!input_init(SMALLINPUT_MODE_NORMAL,0,0)) {
+    puts("Failed to initialize input. Aborting.");
+    return 1;
+  }
   SFG_init();
 
   screen[SCREENSIZE - 1] = 0; // string terminator
